src/lib/mdns/ServiceNaming.cpp

```diff
@@ -119,5 +119,61 @@ CHIP_ERROR MakeHostName(char * buffer, size_t bufferLen, const chip::ByteSpan &
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR MakeServiceSubtype(char * buffer, size_t bufferLen, DiscoveryFilter subtype)
+{
+    size_t requiredSize;
+    switch (subtype.type)
+    {
+    case DiscoveryFilterType::kShort:
+        // 8-bit number
+        if (subtype.code >= 1 << 8)
+        {
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+        requiredSize = snprintf(buffer, bufferLen, "_S%03u", subtype.code);
```

Comment:
Have to remove leading zeroes everywhere

Reply:
Please see comment in commit / above - that should go in as a separate PR with full announcement as it breaks compatibility between controllers and devices compiled against different versions of the SDK.